src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -197,6 +208,26 @@ library BeaconChainProofs {
         require(Merkle.verifyInclusionSha256(proof, beaconStateRoot, balanceRoot, balanceIndex), "BeaconChainProofs.verifyValidatorBalance: Invalid merkle proof");
     }
 
+    function verifySlotRoot(
+        bytes32 beaconStateRoot,
+        bytes32 slotRoot,
+        bytes calldata proof
+    ) internal view {
+        require(proof.length == 32 * (BEACON_STATE_FIELD_TREE_HEIGHT), "BeaconChainProofs.verifySlotRoot: Proof has incorrect length");
+        //Next we verify the slot against the blockHeaderRoot
+        require(Merkle.verifyInclusionSha256(proof, beaconStateRoot, slotRoot, BEACON_STATE_SLOT_INDEX), "BeaconChainProofs.verifyWithdrawalProofs: Invalid slot merkle proof");
+    }
+
+    function verifyStateRootAgainstLatestBlockHeaderRoot(
+        bytes32 beaconStateRoot,
+        bytes32 latestBlockHeaderRoot,
+        bytes calldata proof
+    ) internal view {
+        require(proof.length == 32 * (BEACON_STATE_FIELD_TREE_HEIGHT), "BeaconChainProofs.verifyStateRootAgainstLatestBlockHeaderRoot: Proof has incorrect length");
+        //Next we verify the slot against the blockHeaderRoot
```

Comment:
both this and the logic in the above function can't be doing the exact same thing. here we are verifying the header root against the state root?

Reply:
They are doing the same thing, both the slot and the latest block header are in the beacon state, at the top most level.  So the proofs are gonna have to be the same length

Reply:
> They are doing the same thing, both the slot and the latest block header are in the beacon state, at the top most level. So the proofs are gonna have to be the same length

I was referring to the identical (copy-pasted) comments 😄

Reply:
fixed